gemini/prompts/prompt_optimizer/vapo_lib.py

```diff
@@ -660,3 +670,189 @@ def get_container(self) -> widgets.Output:
                 self.results_output,
             ]
         )
+
+
+def get_id(length: Union[int, None] = 8) -> str:
+    """Generate a uuid of a specified length (default=8)."""
+    if length is None:
+        length = 8
+    return "".join(random.choices(string.ascii_lowercase + string.digits, k=length))
+
+
+def get_auth_token() -> None:
+    """A function to collect the authorization token"""
+    try:
+        result = subprocess.run(
+            ["gcloud", "auth", "print-identity-token", "-q"],
+            capture_output=True,
+            text=True,
+            check=True,
+        )
+        return result.stdout.strip()
+    except subprocess.CalledProcessError as e:
+        print(f"Error getting auth token: {e}")
+        return None
```

Comment:
There should be a way to do this without running the gcloud cli command

Reply:
I agree. I tried the auth module of the google lib but I get some unexpected behaviour when I tried to call the cloud function I am using the token for.